map/bookmark_manager.cpp

```diff
@@ -2976,6 +2977,132 @@ UserMarkLayer * BookmarkManager::GetGroup(kml::MarkGroupId groupId)
   return it != m_categories.end() ? it->second.get() : nullptr;
 }
 
+struct QuadtreeElement
+{
+  int m_rank;
+  Bookmark * m_bm;
+  uint64_t m_zCurveCoord = 0; // z-order curve coordiante
+
+  bool operator < (QuadtreeElement const & rhs) const
+  {
+    return m_zCurveCoord < rhs.m_zCurveCoord;
+  }
+
+  static bool rankLess(QuadtreeElement const & lhs, QuadtreeElement  const & rhs)
+  {
+    return lhs.m_rank < rhs.m_rank;
+  }
+};
+
+using Quadtree = std::vector<QuadtreeElement>;
+
+static void SetMinZoom(Quadtree & quadtree)
```

Comment:
tests for this function?

Reply:
also, this looks heavy enough to merit a separate class or even a separate file

Reply:
> also, this looks heavy enough to merit a separate class or even a separate file

Where is the place for the function (cleaned up from concrete types like `Bookmark` for sure)? Near `scales::`?

Reply:
I see that now it is in kml/ which is fine but you may be interested in another solution to a similar problem in indexer/displacement_manager.hpp